lexer/stateful/stateful.go

```diff
@@ -64,8 +66,151 @@ type compiledRule struct {
 	RE     *regexp.Regexp
 }
 
+// compiledRuleGroup is a series of rules who have been grouped together to form a lexer state.
+// They are analyzed so that for a given input, the rule group may only test the most likely
+// regexps that may match the input.
+type compiledRuleGroup struct {
+	rules []compiledRule
+
+	// These two variables contain information about which rules may be tested on a given rune.
+	// This is done so by having a [nb_runes]bool slice at the end which when true means that the rule at this
+	// index is a possible candidate.
+	//
+	// The rune map is a slice of 256 elements to slices of 256 elements to a slice of nb_rules booleans.
+	//                                              ^ this one can be nil       ^ this one can also be nil
+	// The fallback map is a list of rules that weren't put in the runeMap because if they were they would
```

Comment:
This comment seems like it should be above `fallbackMap`?

Reply:
This comment is in fact outdated and irrelevant since I'm doing a map[]. There are no more 256 buckets